src/duration/mod.rs

```diff
@@ -0,0 +1,356 @@
+//! This module contains a common [`Duration`] struct which is able to parse
+//! human-readable duration formats, like `2y 2h 20m 42s` or`15d 2m 2s`. It
+//! additionally implements many required traits, like [`Derivative`],
+//! [`JsonSchema`], [`Deserialize`][serde::Deserialize], and
+//! [`Serialize`][serde::Serialize].
+//!
+//! Furthermore, it implements [`Deref`], which enables us to use all associated
+//! functions of [`std::time::Duration`] without re-implementing the public
+//! functions on our own type.
+//!
+//! All operators should opt for [`Duration`] instead of the plain
+//! [`std::time::Duration`] when dealing with durations of any form, like
+//! timeouts or retries.
+
+use std::{
+    fmt::{Display, Write},
+    num::ParseIntError,
+    ops::{Deref, DerefMut},
+    str::FromStr,
+};
+
+use derivative::Derivative;
+use schemars::JsonSchema;
+use strum::IntoEnumIterator;
+use thiserror::Error;
```

Comment:
thiserror?

Reply:
We already use it in other places in `operator-rs`. I opted to use `thiserror`, because we want to move in that direction anyways.

Reply:
> because we want to move in that direction anyways.

We have some legacy stuff that uses it, but I'm not aware of a movement to inflict _more_ of it?

Reply:
Changed in 58b04c3